### ISSUE TITLE
feat: Angular 14 Standalone Components

### ISSUE THREAD

**Issue by sean-perkins** [ionic-team/ionic-framework#25404]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already include this feature request, without success.

### Describe the Feature Request

Angular 14 introduces the ability to use standalone components, without having to declare them on an `NgModule`. This feature is used for rendering components within a parent component, as well as for lazy loading a standalone component on a route, instead of using a routing module. 

Standalone components encompass all declarations (pipes, directives and components). 

As of Angular 14, standalone components are a **dev preview** feature. They are also not backwards compatible with older versions of Angular. This means for Ionic to add support for standalone components, Ionic would have to remove support for Angular 13 and lower. 

If you are wanting to _experiment_ with an Angular 14 project and Ionic using standalone components, the following dev-build is available:

```
6.1.7-dev.11654279011.1b971e3e
```

For an example Ionic application on Angular 14, you can clone: https://github.com/sean-perkins/angular-14-demo. Remember to install the available dev-build.

<img width="1440" alt="Screen Shot 2022-06-03 at 2 02 40 PM" src="https://user-images.githubusercontent.com/13732623/171921026-d7b5f180-9528-4529-b2ed-cbea5c118053.png">

### Describe the Use Case

Ionic developers should be able to use the following code:
```ts
{
   path: 'standalone',
   loadComponent: () => import('./standalone.component').then(m => m.StandaloneComponent)
 }
```

To create a route of `/standalone` that renders the `StandaloneComponent` inside of the `ion-router-outlet`.

**Comment by ntorrey** [comment]
Any idea when standalone components will be fully supported?

**Comment by mackelito** [comment]
When will we be able to use standalone components? is there a rough ETA?

**Comment by sean-perkins** [comment]
@mackelito the experimental API support will be available in the next minor release (6.3.0).

This issue is now resolved, since the PR has been merged into the feature branch: https://github.com/ionic-team/ionic-framework/pull/25589

**Comment by hakimio** [comment]
@sean-perkins can you also incorporate PR #25502 in 6.3? I think it's been more than enough time for people to try it out.

**Comment by sean-perkins** [comment]
@hakimio we have a fixed scope agreed upon by the team for 6.3. Standalone component routing I believe was the last task blocking the release. 

The issue you referenced does have some remaining work to get it up-to-date with `main` and verify it against our new test app structure (we are now able to test our changes against v12, v13 and v14 of Angular). 

Assuming all of that goes without issue, we could likely release that fix as a patch release. Our patch release cycle is much more aggressive, releasing weekly if there are changes to go out.
